src/libhoney.js

```diff
@@ -118,10 +118,17 @@ export default class Libhoney extends EventEmitter {
   }
 
   _responseCallback(responses) {
-    let queue = this._responseQueue;
-    if (queue.length < this._options.maxResponseQueueSize) {
-      this._responseQueue = this._responseQueue.concat(responses);
+    const [queue, limit] = [
+      this._responseQueue,
+      this._options.maxResponseQueueSize
+    ];
+
+    if (queue.length < limit) {
+      this._responseQueue = this._responseQueue.concat(
+        concatWithMaxLimit(queue, responses, limit)
+      );
```

Comment:
because this looks very wrong: `concatWithMaxLimit` is supposed to do a concat.  it doesn't.

Reply:
Ah I misunderstood and thought you were wanting to avoid any else statements. If we move all the logic, we'll still have to handle both of those conditions (new responses fit, new responses don't fit but max hasn't been reached). Should have a change up shortly...